### PR TITLE
fix: use assetsDir in creating vite config

### LINKED ISSUE #10626: Worker scripts not output in correct assets directory
### Astro Info

```block
Astro                    v4.5.12
Node                     v18.18.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

The Astro build is picking up the Worker script file and transforming it and placing it in the `dist` directory during builds, but it is always placed in `dist/assets` regardless of what the `build.assets` configuration is set to. 

### What's the expected result?

I would expect the worker script to be output to the same directory as other client scripts as specified by the `build.assets` property in the configuration.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-knx6rb?file=src%2Fpages%2Findex.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
## Changes

fixes #10626 

Sets `assetsDir` (if) `assets` property is passed by user in their Astro configuration.

## Testing

Added tests to match the `custom_dir_1` absolute path to match with the generated directory's `custom_dir_1`.

## Docs

It's a bug so I do not think this needs to be updated, as it now behaves as it was expected to.

### CODE REVIEW

File: packages/astro/src/core/create-vite.ts [hunk 1]
```diff
@@ -239,6 +239,8 @@ export async function createVite(
 	// We also need to filter out the plugins that are not meant to be applied to the current command:
 	// - If the command is `build`, we filter out the plugins that are meant to be applied for `serve`.
 	// - If the command is `dev`, we filter out the plugins that are meant to be applied for `build`.
+	const assetsDir = settings.config.build.assets;
+	if (assetsDir) result = vite.mergeConfig(result, { build: { assetsDir } });
```
Comment (bluwy) [inline packages/astro/src/core/create-vite.ts:243]: Is there a reason why we only set when `build.assets` is specified? Shouldn't we always pass it?

File: .changeset/afraid-laws-speak.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": minor
+---
+
+set build.assets directory during vite config setup
```
Comment (bluwy) [inline .changeset/afraid-laws-speak.md:5]: Do you see some potential behaviour changes that this is marked a minor? If everything else still works as expected, this seems like a patch / bug fix to me.

Comment (rishi-raj-jain) [reply .changeset/afraid-laws-speak.md:5]: Oh right, per https://github.com/withastro/astro/pull/10705#pullrequestreview-1986028561 so the docs mention it it's a bug fix. Fixing.

**Comment by rishi-raj-jain** [comment]
Ready for your review, @bluwy.

File: .changeset/afraid-laws-speak.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+set build.assets directory during vite config setup
```
Comment (ematipico) [inline .changeset/afraid-laws-speak.md:5]:
```suggestion
Correctly sets `build.assets` directory during `vite` config setup
```

File: packages/astro/test/astro-assets-dir.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+						'astro:build:done': ({ dir }) => {
+							checkDir = dir;
+						},
+					},
+				},
+			],
+		});
+		await fixture.build();
+	});
+	it('generates the assets directory per the build.assets configuration', async () => {
```
Comment (ematipico) [inline packages/astro/test/astro-assets-dir.test.js:27]:
```suggestion
	it('generates the assets directory as per build.assets configuration', async () => {
```

**Review by ematipico — APPROVED** [review]

**Review by bluwy — APPROVED** [review]

**Pull request merged**